src/dos/dos_mscdex.cpp

```diff
@@ -290,44 +279,38 @@ int CMscdex::RemoveDrive(uint16_t _drive)
 	return 1;
 }
 
-static CDROM_Interface *create_cdrom_interface(const char *path, uint16_t num_drives)
+static std::unique_ptr<CDROM_Interface> create_cdrom_interface(const char *path)
 {
-	CDROM_Interface *cdrom_interface = nullptr;
 	if (!path_exists(path)) {
-		return nullptr;
+		return {};
 	}
 
 	if (!is_directory(path)) {
-		cdrom_interface = new CDROM_Interface_Image(num_drives);
-		if (cdrom_interface->SetDevice(path)) {
+		if (auto cdrom_interface = std::make_unique<CDROM_Interface_Image>();
+		                                   cdrom_interface->SetDevice(path)) {
```

Comment:
Really not a fan of these multi-statement (with a semi-colon in the middle) if's.  I'm actually not even sure how they work.  Do both statements have to be true (pointer is not null and `SetDevice` returns true)?  Does it only check the last one?

I've seen @johnnovak do this same pattern so if he's OK with it then I'll live.

Reply:
I'm ok with them, and they're really simple:

```cpp
if (<initialisation expression>; <evaluation expression>)
```

E.g. this will execute the expression in the `if` block:

```cpp
if (const int a = 5; a > 1) 
```

Reply:
Makes sense.  You can leave it in if you want then.

Reply:
Also see here @weirddan455. IMO it's a neat C++17 feature:

> C++17 has extended existing if statement’s syntax. Now it is possible to provide initial condition within if statement itself. This new syntax is called "if statement with initializer". This enhancement simplifies common code patterns and helps users keep scopes tight. Which in turn avoids variable leaking outside the scope.


https://stackoverflow.com/a/56866523
https://www.tutorialspoint.com/cplusplus17-if-statement-with-initializer

Reply:
Oh it's a relatively new feature.  That's why I'm just not used to seeing it.  It used to be you couldn't even declare a variable inside an if statement or it would be a compile error.